### PR TITLE
Posix memalign alloc

### PR DESCRIPTION
**Pull request by kouamdo**
Please have a look

### CODE REVIEW

**Comment by kouamdo** [comment]
i have made some modification in each test and power of two function

**Comment by kouamdo** [comment]
I have remove some illogical est in aligned_alloc_test.c 

Please can you have a look ?

File: src/compat/posix/posix_memalign.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+*/
+int power_of_two(int nmber) {
+    if (nmber & (nmber - 1 ) == 0) {
+        return 0 ;
+    } else {
+<<<<<<< HEAD
+        return 1 ;
+=======
+       return 1 ;
+>>>>>>> 72fff1a28...  posix_memalign,aligned_alloc,valloc and pvalloc implementation and with testing
```
Comment (alexkalmuk) [inline src/compat/posix/posix_memalign.c:24]: It looks like some conflicts when you were doing pull or rebase. Can you please fix this?

**Comment by kouamdo** [comment]
I have add some prototype in some header file , ...

**Comment by kouamdo** [comment]
Please , can you have a look ??
i think that i have forget nothing

**Comment by kouamdo** [comment]
`ld -r -o build/base/obj/module/embox/test/stdlib/memmove_test.o -m elf_i386 \`
`./build/base/obj/src/tests/stdlib/memmove_test.o`

`ld: cannot find ./build/base/obj/src/tests/stdlib/memmove_test.o: No such file or directory`

`mk/image2.mk:157: recipe for target 'build/base/obj/module/embox/test/stdlib/memmove_test.o' failed`

`make[4]: *** [build/base/obj/module/embox/test/stdlib/memmove_test.o] Error 1`

this error comming from where ?? Maybe it is making all my changes bad

File: src/tests/mem/pvalloc_test.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    uint32_t i=0;
+    for (i=0;i<_const_size_;i++)mem[i] = (int*)pvalloc(0xF00);
+    for (i=0;i<_const_size_;i++)test_assert_not_equal(mem[i], NULL);
+}
+TEST_CASE("TEST WITH MULTIPLE ALLOCATION WITH LONG SIZE") {
+    int *mem[_const_size_]={NULL} ;
+    uint32_t i=0;
+    for (i=0;i<_const_size_;i++)mem[i] = (int*)pvalloc(i*0xF00);
+    for (i=0;i<_const_size_;i++)test_assert_not_equal(mem[i], NULL);
+    for (i=0;i<_const_size_;i++)free(mem[i]);
```
Comment (alexkalmuk) [inline src/tests/mem/pvalloc_test.c:35]: Fix coding style - spaces, for {, etc

https://github.com/embox/embox/wiki/Code-Style

**Pull request closed without merging**